qaframework-bdd-tests/src/test/java/org/openmrs/contrib/qaframework/automation/VisitSteps.java

```diff
@@ -121,5 +134,71 @@ public void clickOnTheEndVisitButton() {
     @Then("the system ends the visit")
     public void systemEndsVisit() {
     	assertTrue(textExists("No active visit"));
-    }   
+    }
+
+    @When("user clicks on the Active Visits app from the home page")
+    public void loadActiveVisitsPage() {
+    	activeVisitsPage = homePage.goToActiveVisitsSearch();
+    }
+
+    @And("user searches for the patient with active visit using patient Identifier")
+    public void searchForPatientUsingPatientIdentifier() {
+    	activeVisitsPage.search(patient.identifier);
+    	activeVisitsPage.waitForPageToLoad();
+    }
+
+    @And("the system returns patient with the provided patient Identifier")
+    public void systemReturnsPatientWithProvidedIdentifier() {
+    	assertThat(activeVisitsPage.getPatientIdOfLastActiveVisit(), containsString(patient.identifier));
+    }
+
+    @And("user clicks on the patient returned with active visit")
+    public void loadPatientDashboardPage() {
+    	dashboardPage = activeVisitsPage.goToPatientDashboardOfLastActiveVisit();
+    }
+
+    @Then("the system loads the patient dashboard")
+    public void systemLoadsPatientDashboard() {
+    	assertPage(dashboardPage.waitForPage());
+    	dashboardPage.goToHomePage();
+    }
+
+    @And("user searches for the patient with active visit using patient Id")
+    public void searchForPatientUsingPatientId() {
+    	String patientId = activeVisitsPage.getPatientIdOfLastActiveVisit();
+    	activeVisitsPage.search(patientId);
+    	activeVisitsPage.waitForPageToLoad();
+    }
+
+    @And("the system returns patient with the provided patient Id")
+    public void systemReturnsPatientWithProvidedId() {
+    	String patientId = activeVisitsPage.getPatientIdOfLastActiveVisit();
+    	assertThat(activeVisitsPage.getPatientIdOfLastActiveVisit(), is(equalTo(patientId)));
+    }
+
+    @And("user searches for the patient with active visit using patient name")
+    public void searchForPatientUsingPatientName() {
+    	String patientName = activeVisitsPage.getPatientNameOfLastActiveVisit();
+    	activeVisitsPage.search(patientName);
+    	activeVisitsPage.waitForPageToLoad();
+    }
+
+    @And("the system returns patient with the provided patient name")
+    public void systemReturnsPatientWithProvidedName() {
+    	String patientName = activeVisitsPage.getPatientNameOfLastActiveVisit();
+    	assertThat(activeVisitsPage.getPatientNameOfLastActiveVisit(), is(equalTo(patientName)));
+    }
+
+    @And("user searches for the patient with active visit using last seen value")
+    public void searchForPatientUsingLastSeenValue() {
+    	String lastSeen = activeVisitsPage.getPatientLastSeenValueOfLastActiveVisit();
```

Comment:
Could you let me know what `lastSeen` variable represent?

Reply:
@kdaud I am not so sure what values  could be under that lastseen section. I am thinking it could be date or place. I have tried to start a visit for a patient though I don't seem to see anything appearing under that section.But it is one of the test criteria.

Reply:
Could you point me to the actual code in the legacy test?

Reply:
@kdaud here it is in the[ legacy test]( https://github.com/openmrs/openmrs-contrib-qaframework/blob/debbf863ab229daa772f225a4e524e664e5c4bc6/qaframework-legacy-tests/src/test/java/org/openmrs/contrib/qaframework/legacy/SearchActiveVisitsTest.java#L52)